lib/libshout-idjc/src/shout.c

```diff
@@ -0,0 +1,1525 @@
+
+    return _shout_util_dict_get(self->audio_info, name);
+}
+
+int shout_set_meta(shout_t *self, const char *name, const char *value)
+{
+    size_t i;
+
+    if (!self || !name)
+        return SHOUTERR_INSANE;
+
+    if (self->state != SHOUT_STATE_UNCONNECTED)
+        return self->error = SHOUTERR_CONNECTED;
+
+    for (i = 0; name[i]; i++)
+        if ((name[i] < 'a' || name[i] > 'z') && (name[i] < '0' || name[i] > '9'))
+            return self->error = SHOUTERR_INSANE;
+
+    return self->error = _shout_util_dict_set(self->meta, name, value);
+}
+
+const char *shout_get_meta(shout_t *self, const char *name)
+{
+    if (!self)
+        return NULL;
+
+    return _shout_util_dict_get(self->meta, name);
+}
+
+int shout_set_public(shout_t *self, unsigned int public)
+{
+    if (!self || (public != 0 && public != 1))
+        return SHOUTERR_INSANE;
+
+    if (self->state != SHOUT_STATE_UNCONNECTED)
+        return self->error = SHOUTERR_CONNECTED;
+
+    self->public = public;
+
+    return self->error = SHOUTERR_SUCCESS;
+}
+
+unsigned int shout_get_public(shout_t *self)
+{
+    if (!self)
+        return 0;
+
+    return self->public;
+}
+
+int shout_set_format(shout_t *self, unsigned int format)
+{
+    if (!self)
+        return SHOUTERR_INSANE;
+
+    if (self->state != SHOUT_STATE_UNCONNECTED)
+        return self->error = SHOUTERR_CONNECTED;
+
+    if (format != SHOUT_FORMAT_OGG && format != SHOUT_FORMAT_MP3 &&
+        format != SHOUT_FORMAT_WEBM && format != SHOUT_FORMAT_WEBMAUDIO &&
+        format != SHOUT_FORMAT_AAC && format != SHOUT_FORMAT_AACPLUS) {
+        return self->error = SHOUTERR_UNSUPPORTED;
+    }
+
+    self->format = format;
+
+    return self->error = SHOUTERR_SUCCESS;
+}
+
+unsigned int shout_get_format(shout_t* self)
+{
+    if (!self)
+        return 0;
+
+    return self->format;
+}
+
+int shout_set_mimetype(shout_t *self, const char *mimetype)
+{
+    if (!self)
+        return self->error = SHOUTERR_INSANE;
+
+    if (self->state != SHOUT_STATE_UNCONNECTED)
+		return self->error = SHOUTERR_CONNECTED;
+
+    if (self->mimetype)
+        free(self->mimetype);
+        
+    if (!mimetype)
+        self->mimetype = NULL;
+    else
+        if (! (self->mimetype = _shout_util_strdup (mimetype)))
+            return self->error = SHOUTERR_MALLOC;
+        
+    return self->error = SHOUTERR_SUCCESS;
+}
+
+const char *shout_get_mimetype(shout_t *self)
+{        
+    const char *mimetype;
+    
+    if (!self)
+        return NULL;
+        
+    if (self->mimetype)
+        return self->mimetype;
+        
+    switch (self->format) {
+    case SHOUT_FORMAT_OGG:
+        mimetype = "application/ogg";
+        break;
+    case SHOUT_FORMAT_MP3:
+        mimetype = "audio/mpeg";
+        break;
+    case SHOUT_FORMAT_WEBM:
+        mimetype = "video/webm";
+        break;
+    case SHOUT_FORMAT_WEBMAUDIO:
+        mimetype = "audio/webm";
+        break;
+    case SHOUT_FORMAT_AAC:
+        mimetype = "audio/aac";
```

Comment:
This is the die crucial line that allows audio/aac

Reply:
missing here: https://gitlab.xiph.org/xiph/icecast-libshout/-/blob/master/src/shout.c#L927